logr.go

```diff
@@ -258,7 +258,15 @@ type Logger struct {
 // Enabled tests whether this Logger is enabled.  For example, commandline
 // flags might be used to set the logging verbosity and disable some info logs.
 func (l Logger) Enabled() bool {
-	return l.sink != nil && l.sink.Enabled(l.level)
+	return l.sink != nil && l.enabled()
+}
+
+// enabled ensures that Enabled and Info both call LogSink.Enabled with one
+// intermediate stack frame. This wouldn't be necessary if Info had called
+// LogSink.Enabled directly in logr v1.0.0, but it didn't and now that cannot
+// be changed anymore because implementations depend on this behavior.
```

Comment:
I'm not sure I understsand the "implementations depend on this behavior" point.  It seems like we could either do

    (Info, Error, Enabled) -> sink.Enabled

or

    (Info, Error, Enabled) -> enabled -> sink.Enabled

And it would be correct, as long as it is consistent - right?

Reply:
> I'm not sure I understsand the "implementations depend on this behavior" point. 

I mean this code:

https://github.com/kubernetes/klog/blob/6632ba5cc9a5331c853a2c6dbc0fbb125e6e7b94/klogr.go#L55-L56

That code works with logr v1.2.0 for `logger.Info`.

If we switch to `(Info, Error, Enabled) -> sink.Enabled`, `logger.Info` will not work correctly in combination with `-vmodule` anymore when someone uses klog <= v2.100.1 and updates to a future logr release where we made that change.

We can change klog in v2.200.0 and make it depend on such a new logr release, but we cannot prevent the combination above.

> And it would be correct, as long as it is consistent - right?

Both are "correct", but `(Info, Error, Enabled) -> enabled -> sink.Enabled` will cause less pain for downstream users.

Reply:
But then we are left with this weird document that Enabled() needs one more frame. I hate it. If it's already buggy, how bad is it really to just fix the big?